Issue 138: Tighten language around data usage (#141)
epinowcast/baselinenowcast: vignettes/model_definition.Rmd

is the number of cases still missing after $d$ delay. We refer to $X_t$ to describe a random variable, $x_t$ for the corresponding observation, and $\hat{x}_t$ for an estimated/imputed value. The matrix of $x_{t,d}$ available at a given time $t^*$ is referred to as a reporting matrix. In the case where all $t+d>t^*$ have yet to be observed (e.g. $t^*$ is the current time), this reporting matrix is referred to as the reporting triangle, with all values in the bottom right corner of the triangle being missing, except for the first entry at $x_{t=t*, d = 0}$.

> |           | $d = 0$       | $d = 1$       | $d=2$          | $...$ | $d= D-1$         | $d= D$        |

Reviewer: are these all just random whitespace changes?

We describe a method which generates retrospective reporting triangles to replicate what would have been available as of time $t^*=s^*$, where $s^* = t^*-m$ for $m = 1, 2, ... M$ to generate $M$ retrospective reporting triangles.

> To generate the set of $M$ reporting triangles, we work backwards from most recent to oldest, removing the last $m$ rows of the existing reporting triangle, to generate $M$ truncated incomplete reporting matrices. These can be generated from a reporting triangle with the function `truncate_triangles()` which ingests a single reporting triangle and returns a list of `n` truncated reporting matrices containing only observations and missing values if present. We then replace the bottom right triangle of each matrix with NAs, assuming these would not have been observed as of $s^*$, using the function `generate_triangles()` which returns a list of `n` retrospective reporting triangles, again in order from most recent to oldest.

Reviewer: I thought we got rid of the incomplete language and where using truncated vs true reporting triangle only?

Author: Good catch, will go through and remove any references to `incomplete` and matrices that are incorrect in this model definition

The method uses each retrospective reporting triangle to re-estimate a delay distribution using the $N$ preceding reference times of the retrospective reporting triangle before $s^*$, and recomputes a retrospective point nowcast matrix, for $M$ realizations of the retrospective reporting triangle (so $M$ different $s^*$ values).

Thus in order to estimate uncertainty using $N$ reference times, the total training volume must meet or exceed $N+M$ so that the oldest retrospective nowcast dataset at $s^* = t^*-M$ can use $N$ reference times for its point nowcast.

## Generation of retrospective point nowcast matrices

From the $M$ reporting triangles, we apply the method described above to estimate a delay distribution from a reporting triangle and generate a point nowcast for each reporting triangle, to generate $M$ point nowcasts. The function `generate_point_nowcasts()` ingests the list of reporting triangles, estimates a delay distribution for each, and generates a list of point nowcast matrices.

## Fit predicted point nowcast vectors and observed counts to a negative binomial observation model at each forecast horizon

> We choose to quantify the uncertainty at the level of the target quantity which is in this case assumed to be the final count at each reference time, summed across reporting delays.

Reviewer: ```suggestion
We quantify the uncertainty at the target quantity level, which is assumed, by default, to be the final count at each reference time, summed across reporting delays.
```
